libcnb-data/CHANGELOG.md

```diff
@@ -9,6 +9,7 @@
 - Switch `Buildpack.version` from type `semver::Version` to `BuildpackVersion`, in order to validate versions more strictly against the CNB spec ([#241](https://github.com/Malax/libcnb.rs/pull/241)).
 - All libcnb-data struct types now reject unrecognised fields when deserializing ([#252](https://github.com/Malax/libcnb.rs/pull/252)).
 - `BuildpackToml` has been replaced by `BuildpackDescriptor`, which is an enum with `Single` and `Meta` variants that wrap new `SingleBuildpackDescriptor` and `MetaBuildpackDescriptor` types. The new types now reject `buildpack.toml` files where both `stacks` and `order` are present ([#248](https://github.com/Malax/libcnb.rs/pull/248)).
+- Implement `Borrow<str>` for types generated using the `libcnb_newtype!` macro, which allows them to be used with `.join()` ([#258](https://github.com/Malax/libcnb.rs/pull/258)).
```

Comment:
Is this a good entry for the end-user? They would have no visibility which types are created via that macro, only be reading the source.

Reply:
In addition, do we want entry for `libcnb` too since most people use `libcnb-data` implicitly?

Reply:
> In addition, do we want an entry for the libcnb CHANGELOG too since most people use libcnb-data implicitly?

It will be added later, see:
https://github.com/Malax/libcnb.rs/pull/256#issuecomment-1009829942

Reply:
> Is this a good entry for the end-user? They would have no visibility which types are created via that macro, only be reading the source.

Do you have a suggestion as to what else to put? I don't think we should list every single type.

Reply:
Actually, there are only four such types at the moment (I could have sworn there were more? 😆), so I'll list them (`BuildpackId`, `LayerName`, `ProcessType` and `StackId`).